src/player.cpp

```diff
@@ -33,22 +34,101 @@ void Player::setStateChangedCallback(void (*stateChangedCallback)(unsigned int))
     soloud.setStateChangedCallback(stateChangedCallback);
 }
 
-PlayerErrors Player::init(unsigned int sampleRate, unsigned int bufferSize, unsigned int channels)
+PlayerErrors Player::init(unsigned int sampleRate, unsigned int bufferSize, unsigned int channels, int deviceID)
 {
     if (mInited)
         return playerAlreadyInited;
 
+    void *playbackInfos_id = nullptr;
+    if (deviceID != -1)
+    {
+        // Calling this will init [pPlaybackInfos]
+        auto const devices = listPlaybackDevices();
+        if (devices.size() == 0 || deviceID >= devices.size())
+            return backendNotInited;
```

Comment:
Shouldn't this be a better error? Something like `noPlaybackDevicesFound` for the empty list, and `playbackDeviceIdOutOfRange` for the other condition?

Reply:
added CPP `noPlaybackDevicesFound` and counterpart Dart `SoLoudNoPlaybackDevicesFoundCppException`.

Reply:
What about the other error, where the user is asking for a device ID that doesn't exist? (`deviceID >= devices.size()`)? Shouldn't that have a separate error?

Reply:
I was thinking that either there are no devices or you are asking to use one that does not exist (maybe because the BT headphones have been disconnected), the device you would like to use does not exist, and therefore `noPlaybackDevicesFound` would be enough?
Do you think `playbackDeviceIdOutOfRange` should be there too?